<?xml version="1.0"?>
<launch>
  <arg name="ns"            default="stereo_pair"/>
  <arg name="tf_prefix"     default="$(arg ns)"/>
  <arg name="image_left"    default="left/image_rect_color"/>
  <arg name="image_right"   default="right/image_rect_color"/>
  <arg name="markers"       default="markers"/>
  <arg name="stereo_info"   default="stereo_info"/>
  <arg name="display"       default="true"/>

  <arg name="dictionary" default="3"/>
  <arg name="centre"    value="100"/>
  <arg name="top_left"  value="91"/>
  <arg name="top_right" value="92"/>
  <arg name="bot_right" value="108"/>
  <arg name="bot_left"  value="107"/>

  <group ns="$(arg ns)">
      <node name="stereo_detector_node" pkg="aruco_detector" type="stereo_detector_node" output="screen" respawn="true">
        <param name="image_left"    value="$(arg image_left)" />
        <param name="image_right"   value="$(arg image_right)" />
        <param name="markers"       value="$(arg markers)"/>
        <param name="stereo_info"   value="$(arg stereo_info)" />
        <param name="tf_prefix"     value="$(arg tf_prefix)" />
        <param name="display"       value="$(arg display)"/>

        <param name="dictionary" value="$(arg dictionary)"/>
        <param name="centre"    value="$(arg centre)"/>
        <param name="top_left"  value="$(arg top_left)"/>
        <param name="top_right" value="$(arg top_right)"/>
        <param name="bot_right" value="$(arg bot_right)"/>
        <param name="bot_left"  value="$(arg bot_left)"/>
      </node>
   </group>
</launch>
